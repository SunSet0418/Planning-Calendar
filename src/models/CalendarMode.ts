export enum CalendarMode {
    Month = '월',
    Week = '주'
}