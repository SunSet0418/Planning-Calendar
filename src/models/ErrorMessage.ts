export class ErrorMessage {
    constructor(
        public title: string,
        public description: string
    ) { }
}