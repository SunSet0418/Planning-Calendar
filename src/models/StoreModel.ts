import { TaskStoreModel } from '../modules/reducers/task';

export interface StoreModel {
    task: TaskStoreModel
}