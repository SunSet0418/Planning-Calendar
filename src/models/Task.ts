export interface Task {
    id: number
    startDate: string,
    endDate: string,
    startTime: number,
    durationTime: number
}