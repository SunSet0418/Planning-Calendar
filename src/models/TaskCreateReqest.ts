export interface TaskCreateRequest {
    name: string,
    date: string,
    startHour: number,
    endHour: number
}