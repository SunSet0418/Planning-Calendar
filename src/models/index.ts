export { TaskStore } from './TaskStore';
export { Task } from './Task';
export { StoreModel } from './StoreModel';
export { CalendarMode } from './CalendarMode';
export { ErrorMessage } from './ErrorMessage';