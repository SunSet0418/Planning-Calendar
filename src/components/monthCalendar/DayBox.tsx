import * as React from 'react';
import * as moment from 'moment';
import styled from 'styled-components';
import {TaskItem} from '../TaskItem';
import { Task } from '../../models';
import 'antd/lib/button/style/css';

const Day = styled.div`
    text-align: left;
    padding: 10px;
`
Day.displayName = 'Day';

interface Props {
    dateMoment: moment.Moment,
    tasks: Task[]
}

export const DayBox: React.FC<Props> = (props: Props) => {
    const { dateMoment, tasks } = props;
    const sortedTask = tasks.sort((task1: Task, task2: Task) => (task1.startHour >= task2.startHour) ? 1 : -1);
    return (
        <Day data-datetime={dateMoment.format()}>
            {dateMoment.date()}
            <div>
                {sortedTask.map(task => ( <TaskItem key={task.id} taskId={task.id} name={task.name}/> ))}
            </div>
        </Day>
    )
}