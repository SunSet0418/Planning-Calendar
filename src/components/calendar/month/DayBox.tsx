import * as React from 'react';
import * as moment from 'moment';
import styled from 'styled-components';
import { Task } from '../../../models/Task';
import { compareMoment } from '../../../libs/calendarUtil';
import { Button } from 'antd';
import 'antd/lib/button/style/css';

const Day = styled.div`
    text-align: left;
    padding: 10px;
`
const TaskItem = styled(Button)`
    width: 100%;
    height: auto;
`
Day.displayName = 'Day';
TaskItem.displayName = 'Task';

interface Props {
    month: number,
    date: number,
    tasks: Task[]
}

export const DayBox: React.FC<Props> = (props: Props) => {
    const {date, month, tasks} = props;
    const sortedTask = tasks.sort((task1: Task, task2: Task) => compareMoment(moment(task1.date), moment(task2.date)));
    return (
        <Day data-date={date} data-month={month}>
            {date}
            <div>
                {sortedTask.map(task => <TaskItem key={task.id} data-taskid={task.id}>{task.name}</TaskItem>)}
            </div>
        </Day>
    )
}