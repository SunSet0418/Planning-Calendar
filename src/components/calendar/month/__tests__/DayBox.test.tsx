import * as React from 'react';
import * as Enzyme from 'enzyme';
import { Task } from '../../../../models/Task';
import { DayBox } from '../DayBox';

describe('<DayBox />', () => {
    const tasks = [
        new Task(1, '귤 까기10', '2014-02-10T10:00:00.000Z', 1, 1),
        new Task(1, '귤 까기10', '2014-02-10T10:00:00.000Z', 10, 1)
    ];
    let component = null;


    it('renders correctly', () => {
        component = Enzyme.shallow(<DayBox month={2} date={10} tasks={tasks}/>);
    });

    it('should match snapshot', () => {
        expect(component).toMatchSnapshot();
    });

    it('should row exists 5', () => {
        expect(component.find('Task').length).toBe(2);
        expect(component.find('Task').at(0).text()).toBe('귤 까기10');
    });

});