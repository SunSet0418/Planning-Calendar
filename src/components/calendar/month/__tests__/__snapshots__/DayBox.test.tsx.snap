// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DayBox /> should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    date={10}
    month={2}
    tasks={
        Array [
            Task {
              "date": "2014-02-10T01:00:00.000Z",
              "durationTime": 1,
              "id": 1,
              "name": "귤 까기01",
            },
            Task {
              "date": "2014-02-10T09:00:00.000Z",
              "durationTime": 3,
              "id": 1,
              "name": "귤 까기09",
            },
            Task {
              "date": "2014-02-10T10:00:00.000Z",
              "durationTime": 2,
              "id": 1,
              "name": "귤 까기10",
            },
            Task {
              "date": "2014-02-10T12:00:00.000Z",
              "durationTime": 4,
              "id": 1,
              "name": "귤 까기12",
            },
            Task {
              "date": "2014-02-10T24:00:00.000Z",
              "durationTime": 10,
              "id": 1,
              "name": "귤 까기24",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        10,
        <div>
          <Unknown
                    data-taskId={1}
          >
                    귤 까기01
          </Unknown>
          <Unknown
                    data-taskId={1}
          >
                    귤 까기09
          </Unknown>
          <Unknown
                    data-taskId={1}
          >
                    귤 까기10
          </Unknown>
          <Unknown
                    data-taskId={1}
          >
                    귤 까기12
          </Unknown>
          <Unknown
                    data-taskId={1}
          >
                    귤 까기24
          </Unknown>
</div>,
      ],
      "data-date": 10,
      "data-month": 2,
    },
    "ref": null,
    "rendered": Array [
      10,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Unknown
              data-taskId={1}
>
              귤 까기01
</Unknown>,
            <Unknown
              data-taskId={1}
>
              귤 까기09
</Unknown>,
            <Unknown
              data-taskId={1}
>
              귤 까기10
</Unknown>,
            <Unknown
              data-taskId={1}
>
              귤 까기12
</Unknown>,
            <Unknown
              data-taskId={1}
>
              귤 까기24
</Unknown>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "function",
            "props": Object {
              "children": "귤 까기01",
              "data-taskId": 1,
            },
            "ref": null,
            "rendered": "귤 까기01",
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "Group": [Function],
              "__ANT_BUTTON": true,
              "attrs": Array [],
              "componentStyle": ComponentStyle {
                "componentId": "sc-bwzfXH",
                "isStatic": true,
                "rules": Array [
                  "
    width: 100%;
    height: auto;
",
                ],
              },
              "displayName": "Task",
              "foldedComponentIds": Array [],
              "render": [Function],
              "styledComponentId": "sc-bwzfXH",
              "target": [Function],
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
              Symbol(Symbol.hasInstance): [Function],
            },
          },
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "function",
            "props": Object {
              "children": "귤 까기09",
              "data-taskId": 1,
            },
            "ref": null,
            "rendered": "귤 까기09",
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "Group": [Function],
              "__ANT_BUTTON": true,
              "attrs": Array [],
              "componentStyle": ComponentStyle {
                "componentId": "sc-bwzfXH",
                "isStatic": true,
                "rules": Array [
                  "
    width: 100%;
    height: auto;
",
                ],
              },
              "displayName": "Task",
              "foldedComponentIds": Array [],
              "render": [Function],
              "styledComponentId": "sc-bwzfXH",
              "target": [Function],
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
              Symbol(Symbol.hasInstance): [Function],
            },
          },
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "function",
            "props": Object {
              "children": "귤 까기10",
              "data-taskId": 1,
            },
            "ref": null,
            "rendered": "귤 까기10",
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "Group": [Function],
              "__ANT_BUTTON": true,
              "attrs": Array [],
              "componentStyle": ComponentStyle {
                "componentId": "sc-bwzfXH",
                "isStatic": true,
                "rules": Array [
                  "
    width: 100%;
    height: auto;
",
                ],
              },
              "displayName": "Task",
              "foldedComponentIds": Array [],
              "render": [Function],
              "styledComponentId": "sc-bwzfXH",
              "target": [Function],
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
              Symbol(Symbol.hasInstance): [Function],
            },
          },
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "function",
            "props": Object {
              "children": "귤 까기12",
              "data-taskId": 1,
            },
            "ref": null,
            "rendered": "귤 까기12",
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "Group": [Function],
              "__ANT_BUTTON": true,
              "attrs": Array [],
              "componentStyle": ComponentStyle {
                "componentId": "sc-bwzfXH",
                "isStatic": true,
                "rules": Array [
                  "
    width: 100%;
    height: auto;
",
                ],
              },
              "displayName": "Task",
              "foldedComponentIds": Array [],
              "render": [Function],
              "styledComponentId": "sc-bwzfXH",
              "target": [Function],
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
              Symbol(Symbol.hasInstance): [Function],
            },
          },
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "function",
            "props": Object {
              "children": "귤 까기24",
              "data-taskId": 1,
            },
            "ref": null,
            "rendered": "귤 까기24",
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "Group": [Function],
              "__ANT_BUTTON": true,
              "attrs": Array [],
              "componentStyle": ComponentStyle {
                "componentId": "sc-bwzfXH",
                "isStatic": true,
                "rules": Array [
                  "
    width: 100%;
    height: auto;
",
                ],
              },
              "displayName": "Task",
              "foldedComponentIds": Array [],
              "render": [Function],
              "styledComponentId": "sc-bwzfXH",
              "target": [Function],
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
              Symbol(Symbol.hasInstance): [Function],
            },
          },
        ],
        "type": "div",
      },
    ],
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": true,
        "rules": Array [
          "
    text-align: left;
    padding: 10px;
",
        ],
      },
      "displayName": "Day",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          10,
          <div>
            <Unknown
                        data-taskId={1}
            >
                        귤 까기01
            </Unknown>
            <Unknown
                        data-taskId={1}
            >
                        귤 까기09
            </Unknown>
            <Unknown
                        data-taskId={1}
            >
                        귤 까기10
            </Unknown>
            <Unknown
                        data-taskId={1}
            >
                        귤 까기12
            </Unknown>
            <Unknown
                        data-taskId={1}
            >
                        귤 까기24
            </Unknown>
</div>,
        ],
        "data-date": 10,
        "data-month": 2,
      },
      "ref": null,
      "rendered": Array [
        10,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Unknown
                data-taskId={1}
>
                귤 까기01
</Unknown>,
              <Unknown
                data-taskId={1}
>
                귤 까기09
</Unknown>,
              <Unknown
                data-taskId={1}
>
                귤 까기10
</Unknown>,
              <Unknown
                data-taskId={1}
>
                귤 까기12
</Unknown>,
              <Unknown
                data-taskId={1}
>
                귤 까기24
</Unknown>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "function",
              "props": Object {
                "children": "귤 까기01",
                "data-taskId": 1,
              },
              "ref": null,
              "rendered": "귤 까기01",
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "Group": [Function],
                "__ANT_BUTTON": true,
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-bwzfXH",
                  "isStatic": true,
                  "rules": Array [
                    "
    width: 100%;
    height: auto;
",
                  ],
                },
                "displayName": "Task",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-bwzfXH",
                "target": [Function],
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
                Symbol(Symbol.hasInstance): [Function],
              },
            },
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "function",
              "props": Object {
                "children": "귤 까기09",
                "data-taskId": 1,
              },
              "ref": null,
              "rendered": "귤 까기09",
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "Group": [Function],
                "__ANT_BUTTON": true,
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-bwzfXH",
                  "isStatic": true,
                  "rules": Array [
                    "
    width: 100%;
    height: auto;
",
                  ],
                },
                "displayName": "Task",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-bwzfXH",
                "target": [Function],
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
                Symbol(Symbol.hasInstance): [Function],
              },
            },
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "function",
              "props": Object {
                "children": "귤 까기10",
                "data-taskId": 1,
              },
              "ref": null,
              "rendered": "귤 까기10",
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "Group": [Function],
                "__ANT_BUTTON": true,
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-bwzfXH",
                  "isStatic": true,
                  "rules": Array [
                    "
    width: 100%;
    height: auto;
",
                  ],
                },
                "displayName": "Task",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-bwzfXH",
                "target": [Function],
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
                Symbol(Symbol.hasInstance): [Function],
              },
            },
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "function",
              "props": Object {
                "children": "귤 까기12",
                "data-taskId": 1,
              },
              "ref": null,
              "rendered": "귤 까기12",
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "Group": [Function],
                "__ANT_BUTTON": true,
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-bwzfXH",
                  "isStatic": true,
                  "rules": Array [
                    "
    width: 100%;
    height: auto;
",
                  ],
                },
                "displayName": "Task",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-bwzfXH",
                "target": [Function],
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
                Symbol(Symbol.hasInstance): [Function],
              },
            },
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "function",
              "props": Object {
                "children": "귤 까기24",
                "data-taskId": 1,
              },
              "ref": null,
              "rendered": "귤 까기24",
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "Group": [Function],
                "__ANT_BUTTON": true,
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-bwzfXH",
                  "isStatic": true,
                  "rules": Array [
                    "
    width: 100%;
    height: auto;
",
                  ],
                },
                "displayName": "Task",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-bwzfXH",
                "target": [Function],
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
                Symbol(Symbol.hasInstance): [Function],
              },
            },
          ],
          "type": "div",
        },
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": true,
          "rules": Array [
            "
    text-align: left;
    padding: 10px;
",
          ],
        },
        "displayName": "Day",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
